qcodes/instrument_drivers/AlazarTech/dll_wrapper.py

```diff
@@ -0,0 +1,106 @@
+import ctypes
+import logging
+import asyncio
+import concurrent
+import sys
+import re
+from typing import TypeVar, Type, Dict, Tuple, Callable, NamedTuple, Sequence, NewType, List, Any
+from threading import Lock
+from functools import partial
+
+from qcodes.instrument.parameter import Parameter
+from qcodes.instrument_drivers.AlazarTech.utils import TraceParameter
+
+
+logger = logging.getLogger(__name__)
+
+TApi = TypeVar("TApi", bound="AlazarATSAPI")
+ReturnCode = NewType('ReturnCode', ctypes.c_uint)
+
+
+## FUNCTIONS ##
+
+
+def api_call_task(lock, c_func, callback, *args):
+    with lock:
+        retval = c_func(*args)
+    callback()
+    return retval
+
+
+def convert_to_camel_case(name):
+    # https://stackoverflow.com/a/1176023
+    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
+
+def normalize_params(*args) -> List[Any]:
+    args_out: List[int] = []
+    for arg in args:
+        if isinstance(arg, Parameter):
+            args_out.append(arg.raw_value)
+        else:
+            args_out.append(arg)
+    return args_out
+
+
+def mark_params_as_updated(*args) -> None:
+    for arg in args:
+        if isinstance(arg, TraceParameter):
+            arg._set_updated()
+
+
+def api_calls(full_name: str, signature: "Signature") -> Tuple[Callable, Callable]:
+    def sync_call(self: TApi, *args) -> signature.return_type:
+        c_func = getattr(self._dll, full_name)
+        future = self._executor.submit(
+            api_call_task,
+            self._lock, c_func,
+            partial(mark_params_as_updated, *args),
+            *normalize_params(*args)
+        )
+        return future.result()
+
+    def async_call(self: TApi, *args) -> signature.return_type:
```

Comment:
@QCoDeS/core do we follow yagni and remove all traces of async for now (in this PR)?

Reply:
if you mean asyncio then yes I think so. (the driver inherently uses the async alazar c-api and that should not be removed).  If we do not remove asyncio support we will have to sort out the issues pointed out with async in the other pr first.

Reply:
(yes, i meant `asyncio`). ok, will remove `asyncio` for this PR.